client/post-editor/editor-sidebar/constants.js

```diff
@@ -0,0 +1,12 @@
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+
+export const NESTED_SIDEBAR_NONE = 'NESTED_SIDEBAR_NONE';
```

Comment:
Fair point, though at this point I'm not convinced we need it: design-wise, the states we expect for our editor sidebar are: no sidebar (collapsed), the default editor sidebar (corresponds to `NESTED_SIDEBAR_NONE`), the revisions one, and any other we come up with, and the current data structure encodes the fact that there's always a single sidebar (or none) at any given time.

It's a possibility that another structure becomes needed in the future, but let's apply the principle of [YAGNI](https://ddg.gg/yagni) for now.